### ISSUE TITLE
Very strong correlation of predictions between drugs

### ISSUE THREAD

**Issue by raphaelmourad** [ArcInstitute/state#153]
Hi,

given a set of 5000 keratinocyte cells, I have predicted with ST-Tahoe the expression after 2 different treatments:
- neratinib 5 uM
- oleic acid 5 uM

Here are the h5ad files from ST-Tahoe (after prediction), if anyone wants to reproduce my results:
https://transfert.free.fr/u1Sl77T
NB: before running ST-Tahoe, I kept only the 2000 genes from var_dims.pkl 

State command:
state tx infer \
  --model_dir ST-Tahoe \
  --checkpoint ST-Tahoe/final.ckpt \
  --adata examples/$file_se \
  --output examples/$file_pred \
  --pert_col drugname_drugconc \
  --batch_size 64

Predictions were going well:
<img width="1078" height="280" alt="Image" src="https://github.com/user-attachments/assets/a124febe-9d04-446d-8fd9-090b146a042c" />

When I look at the pseudobulk expression of genes (for each gene I average over the 5000 cells after drug):
`mean_exp_neratinib=np.mean(adata_neratinib.X,axis=0)`
`mean_exp_oleic_acid=np.mean(adata_oleic_acid.X,axis=0)`

If I check if the predictions are different, I get:
`np.sum(mean_exp_neratinib==mean_exp_oleic_acid)/len(mean_exp_neratinib)`
np.float64(0.642)

meaning that 36% of predicted values are different between the 2 drugs, which is totally fine since many genes have a mean =0, which would lead to same values between different drugs. 

BUT, when I compute the correlation between the 2 drugs, I get 0.9999999.
<img width="622" height="467" alt="Image" src="https://github.com/user-attachments/assets/56d2c655-29ac-4c89-97c0-2d8b15a78930" />

After excluding mean=0 for both drugs:, I still get a strong correlation, in particular for genes with high expression:

<img width="622" height="467" alt="Image" src="https://github.com/user-attachments/assets/f9d74841-dad5-487a-aa69-642f4e41e0f2" />

Do you have any explanations? Is there a way to remove such correlation between drugs? I feel this is the "mode collapse" phenomenon reported many times as in Perturbench paper.

Regards
Raphaël

**Comment by raphaelmourad** [comment]
Fyi: i didn't log the data before predicting with ST-Tahoe and i just normalized the count with the median as suggested in the issues.

**Comment by abhinadduri** [comment]
it seems like the cells might be your own data from the image - if so, I think you may have more success with: 1) a state model with SE embeddings, pretrained on Tahoe, and 2) some fine tuning on your target data.

we can look into uploading 1) to hugging face, and perhaps making a colab demonstrating how to make 2)

**Comment by ttruan** [comment]
I would be extremely grateful if you could provide the SE fine‑tuned on the target data.

**Comment by raphaelmourad** [comment]
As said above, I didn't finetune anything, I used SE-600M with  se600m_epoch4.ckpt, and ST-Tahoe with  final.ckpt.

**Comment by ttruan** [comment]
Thank you for your reply; I understand. I was just hoping the authors could share some tips on fine‑tuning the SE model—I’ve been working on it for a long time but still haven’t managed to fine‑tune it properly.

**Comment by raphaelmourad** [comment]
@abhinadduri ok thanks! Just saying here i would just love to be able to predict (not train) the effect of a drug on a novel set of cells with anf without embedding

**Comment by Rive-001** [comment]
@raphaelmourad We've added a [Colab Notebook](https://colab.research.google.com/drive/1bq5v7hixnM-tZHwNdgPiuuDo6kuiwLKJ) to demonstrate how to use the ST model to perform inference. 

We tried investigating the correlation between the perturbation effect predictions of a subset of holdout perturbations in the Tahoe dataset. We arrived at this correlation matrix of predicted delta for cell type `C32`. I hope this helps answer your query!

<img width="1261" height="1012" alt="Image" src="https://github.com/user-attachments/assets/bda8be69-126f-43e6-8811-825fc5bd2817" />
